### PR TITLE
code monitoring: update email template

### PR DESCRIPTION
**Pull request by limitedmage**
Part of #31580

Updates code monitoring email templates to match the new format [described in this doc](https://docs.google.com/document/d/1iqpCma3Em6fj4Mkv6PRo-4lszIMMxKTW45PzXqRBDEo/edit).
Conditional results will be added in a subsequent PR.

### Template from test data rendered in a browser
image

### Test email rendered in Gmail

image

## Test plan

- Updated Go unit tests to check that the new template is rendered
- Manual testing of test email

### CODE REVIEW

File: enterprise/internal/codemonitors/background/email.go [hunk 1]
```diff
@@ -41,56 +41,59 @@ var (
 )
 
 var newSearchResultsEmailTemplates = txemail.MustValidate(txtypes.Templates{
-	Subject: `{{ if .IsTest }}Test: {{ end }}[{{.Priority}} event] {{.Description}}`,
+	Subject: `{{ if .IsTest }}Test: {{ end }}{{.Priority}}Sourcegraph code monitor {{.Description}} detected {{.NumberOfResults}} new {{.ResultPluralized}}`,
```
Comment (camdencheek) [inline enterprise/internal/codemonitors/background/email.go:44]: Should there be a space after `{{.Priority}}`?

**Review by camdencheek — APPROVED** [review]

File: enterprise/internal/codemonitors/background/email.go (hunk 1 above)
Comment (limitedmage) [reply enterprise/internal/codemonitors/background/email.go:44]: I changed priority to either be `""` or `"Critical "`, so nothing is rendered in normal priority but we still show critical priority (with a space included).

Comment (camdencheek) [reply enterprise/internal/codemonitors/background/email.go:44]: It looks like it was changed to ` [Critical]`, not `[Critical] `

Comment (camdencheek) [reply enterprise/internal/codemonitors/background/email.go:44]: wow, that did not render usefully. The space is before the priority, not after 😄

Comment (limitedmage) [reply enterprise/internal/codemonitors/background/email.go:44]: Oh you're right! Thanks for the catch. Should be fixed now, with space before and after.
Do we use the critical priority at all right now anyway?

Comment (camdencheek) [reply enterprise/internal/codemonitors/background/email.go:44]: >  Do we use the critical priority at all right now anyway?

now that you say it, I don't think we do. We pass it all over the backend, but we never set it 😄

**Pull request merged**